feature/mpls/sr/otg_tests/srte_egress_node_forwarding/egress_node_forwading_test.go

```diff
@@ -0,0 +1,667 @@
+	case txPkts > 0:
+		t.Logf("Flow %s: Successfully transmitted %d packets and received %d packets with no loss.", flowName, txPkts, rxPkts)
+	case packetsPerFlow > 0: // Expected to send but didn't
+		return 0, 0, fmt.Errorf("flow %s: no packets were transmitted for this flow, but %d were expected", flowName, packetsPerFlow)
+	}
+
+	return totalTxFromATE, totalRxAtATE, nil
+
+}
+
+func sendTrafficCapture(t *testing.T, ate *ondatra.ATEDevice) {
+	t.Logf("Starting capture")
+	cs := startCapture(t, ate.OTG())
+
+	t.Logf("Starting traffic on ATE...")
+	ate.OTG().StartTraffic(t)
+	t.Logf("Waiting for %v for traffic to complete and stats to update...", trafficTimeout)
+	time.Sleep(trafficTimeout)
+	ate.OTG().StopTraffic(t)
+	t.Logf("Traffic stopped.")
+
+	t.Logf("Stop Capture")
+	stopCapture(t, ate.OTG(), cs)
+}
+
+func captureDUTCounters(t *testing.T, dut *ondatra.DUTDevice) (uint64, uint64) {
+	dutP1 := dut.Port(t, "port1")
+	dutP2 := dut.Port(t, "port2")
+
+	// Capture initial DUT interface counters for delta calculation.
+	dutP1InCounters := gnmi.Get(t, dut, gnmi.OC().Interface(dutP1.Name()).Counters().State()).GetInUnicastPkts()
+	dutP2OutCounters := gnmi.Get(t, dut, gnmi.OC().Interface(dutP2.Name()).Counters().State()).GetOutUnicastPkts()
+
+	return dutP1InCounters, dutP2OutCounters
+}
+
+func validateDUTCounters(t *testing.T, flowCount int, dutP1InCountersBefore, dutP1InCountersAfter, dutP2OutCountersBefore, dutP2OutCountersAfter, totalTxFromATE, totalRxAtATE uint64) error {
+	// DUT Counter Validation (Delta Check)
+	inPktsDelta := dutP1InCountersAfter - dutP1InCountersBefore
+	outPktsDelta := dutP2OutCountersAfter - dutP2OutCountersBefore
+
+	t.Logf("DUT Port1 InUnicastPkts: Before=%d, After=%d, Delta=%d", dutP1InCountersBefore, dutP1InCountersAfter, inPktsDelta)
+
+	t.Logf("DUT Port2 OutUnicastPkts: Before=%d, After=%d, Delta=%d", dutP2OutCountersBefore, dutP2OutCountersAfter, outPktsDelta)
+
+	t.Logf("Total packets sent by ATE (sum of flows reported by ATE): %d", totalTxFromATE)
+	t.Logf("Total packets received by ATE (sum of flows reported by ATE): %d", totalRxAtATE)
+
+	expectedTotalTraffic := uint64(packetsPerFlow) * uint64(flowCount)
+	switch {
+	case totalTxFromATE > 0:
+		// Check if DUT ingress counters reflect ATE Tx.
+		// Allow a small difference (e.g., 2%) due to other control plane packets or timing of counter polling.
+		if float64(inPktsDelta) < float64(totalTxFromATE)*0.98 {
+			return fmt.Errorf("dut port1 InUnicastPkts delta (%d) is significantly less than ATE Tx (%d) - expected approx %d", inPktsDelta, totalTxFromATE, expectedTotalTraffic)
+		}
+		// Check if DUT egress counters reflect ATE Rx.
+		if float64(outPktsDelta) < float64(totalRxAtATE)*0.98 {
+			return fmt.Errorf("dut Port2 OutUnicastPkts delta (%d) is significantly less than ATE Rx (%d) - potential drop in DUT. expected approx %d", outPktsDelta, totalRxAtATE, expectedTotalTraffic)
```

Comment:
https://google.github.io/styleguide/go/decisions#error-strings

```suggestion
			return fmt.Errorf("dut Port2 OutUnicastPkts delta (%d) is significantly less than ATE Rx (%d) - potential drop in DUT, expected approx %d", outPktsDelta, totalRxAtATE, expectedTotalTraffic)
```